packages/bot/src/scratch/blocks/Binary/Trade Definition/trade_definition_tradeoptions.js

```diff
@@ -211,6 +213,25 @@ Blockly.Blocks.trade_definition_tradeoptions = {
             }
         });
     },
+    validateDurationInput() {
+        const { contracts_for } = ApiHelpers.instance;
+
+        contracts_for.getDurations(this.selected_symbol, this.selected_trade_type).then(durations => {
+            const duration_block    = this.getInputTargetBlock('DURATION');
+            const duration_input    = duration_block.getFieldValue('NUM');
+            const { min, max }      = durations.find(d => d.unit === this.selected_duration);
+
+            const is_valid_duration = duration_input >= min && duration_input <= max;
+            let error_message = localize(`Duration value is not allowed. To run the bot, please enter a value between ${min} to ${max}.`);
```

Comment:
Please use `localize`'s placeholders rather than template literals. e.g. 

```JavaScript
let error_message = localize(`Duration value is not allowed. To run the bot, please enter a value between {{ min }} to {{ max }}.`, { min, max });
```

As per @mahdipourziaei-binary 

> The string extractor reads all of the code inside src folder and extracts all strings that are inside localize('everything here') and <Localize i18n_default_text='And everything here' /> the localize() or <Localize /> (and Crowdin) do not accept JS string interpolation, so we can’t do
>
> ```localize(`hello i am ${name}`)```
>
> , you write it like this:
>
> ```localize('hello i am {{ name_prop }}', { name_prop: 'John Doe' })```